pkgs/shells/nushell/default.nix

```diff
@@ -21,16 +21,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "nushell";
-  version = "0.65.0";
+  version = "0.66.2";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = version;
-    sha256 = "sha256-KgXhmAOJaAvmNuDqSaW+h6GF5rWYgj8/wn+vz9V9S7M=";
+    sha256 = "sha256-PaPj2hbObY4wXubN3Mvr0TlpI13Zgkey90qAgpB0qOo=";
   };
 
-  cargoSha256 = "sha256-YqtM/1p6oP0+E0rYSFPeCbof06E81eC2PZIwkU7J0I4=";
+  cargoSha256 = "sha256-4gYafGaQu+UWMQoQ9bf8Cm5rKZhO5ZbKDOitC4HXNdI=";
   # Since 0.34, nu has an indirect dependency on `zstd-sys` (via `polars` and
```

Comment:
This whole section (`depsExtraArgs` and `cargoUpdateHook`) isn't really necessary anymore, builds fine without it.

`cargoSha256 = "sha256-4gYafGaQu+UWMQoQ9bf8Cm5rKZhO5ZbKDOitC4HXNdI=";`

And dataframe commands work fine:

![image](https://user-images.githubusercontent.com/20619776/181837102-3f01f37d-5c92-4951-a91a-3402682de84b.png)

Reply:
`zstd-sys` still depends on the Git submodule.

Reply:
The [original complaint](https://github.com/NixOS/nixpkgs/pull/130760#issuecomment-883317581) was that it failed to build without resorting to the `pkg-config` cargo patch, seems like that's not the case anymore.

Reply:
You'd be right, the `zstd-sys` crate now properly includes the `zstd` source without needing to fetch the submodule. However, wouldn't it still be a good idea to use our `zstd` if possible, like it is here?

Reply:
Hmm, maybe, but it isn't worth fighting the build system for it.

Reply:
It's not really fighting the build system if we just have to flip a feature flag, IMO.

Reply:
[This patch](https://gist.github.com/liketechnik/bfa19583cdcd55e6778615dfb17488b7) (unfortunately comment attachments do not support diffs, so linked as a gist) added to `cargoPatches` allows to use the system zstd with nushell 0.66.2. 

It's a (minor) modification of the patch from the original issue linked above. I've changed to location of the zstd dependency, in order to workaround rust-lang/cargo#7880.

Reply:
I see other packages in nixpkgs using this pattern (e.g. `cargo-about` and `cargo-deny`), which seems like enough precedent that I am comfortable with it. I've incorporated @liketechnik 's patch in 1babd46a500ac7a6c2d59a0366a966a406500999. Please let me know if you'd like me to handle the attribution any differently, I'm fine with whatever.

I'd feel best if we hear from @Artturin about it as well?

Reply:
> It's a (minor) modification of the patch from the original issue linked above. I've changed to location of the zstd dependency, in order to workaround [rust-lang/cargo#7880](https://github.com/rust-lang/cargo/issues/7880).

I can't reproduce this issue by adding the exact same line you did to the workspace's `Cargo.toml`, FWIW. (It doesn't really matter, but definitely interesting -- were you able to reproduce it, or were you just adding it there just in case?)

Reply:
Oops. You're observation is most likely correct. Apparently I got confused with multiple attempts to get this working, a) replacing the version number from the original version with a wildcard and b) moving it from the workspace to a specific crate. And I guess a) is the relevant fix and *not* moving it away from the workspace.

Not sure if it should be moved back into the workspace's `Cargo.toml`, then.

Reply:
It doesn't really matter honestly, ideally it would be in the `Cargo.toml` so it's future proof to more `zstd-sys` dependents, but depending on how often upstream changes its dependencies, the patch will probably have to be regenerated at some point so it could be just done then. 

But now that I think about it, the dependency in the `Cargo.toml` should probably be the package which directly consumes the `pkg-config` feature flag, which is `zstd-sys`, as a matter of form/correctness. What do you think, am I just nitpicking here?

(Also, this patch will hopefully be made redundant in the future. When I get a chance, I'll send a patch upstream to allow setting this option, so we won't have to patch it at all for the next version. (Anyone else interested can PR this change upstream, I don't mind -- I'm just also happy to do it.))